### PR TITLE
Allow Symfony 3.x Components

### PR DESCRIPTION
**Pull request by Spomky**
This PR allows Symfony 3.x Components to be used.

### CODE REVIEW

**Comment by prisis** [comment]
@padraic can you merge this?

**Comment by GrahamCampbell** [comment]
Looks good to me. :+1:

**Comment by jayS-de** [comment]
I would suggest to also change the Symfony2 packages from the tilde to the caret operator.

E.g.:

```
"symfony/finder": "^2.6|^3.0"
```

As it should not make a difference in the dependency resolution, the dependencies only use one resolution pattern.

**Comment by Spomky** [comment]
@jayS-de tilde to caret done

File: composer.json [hunk 1]
```diff
@@ -29,7 +29,7 @@
     },
     "require-dev": {
         "mockery/mockery": "^0.9",
-        "behat/behat": "^3.0.15",
+        "behat/behat": "^3.0",
```
Comment (jayS-de) [inline composer.json]: This doesn't change anything. Behat 3.1 was already supported by this constraint.

See https://getcomposer.org/doc/articles/versions.md#caret

The test is breaking for a different reason

**Comment by padraic** [comment]
All checks now passing!

**Pull request merged**